syntax = "proto3";

service RaftKeyValue {
    rpc Get (GetRequest) returns (GetResponse) {}
    rpc Put (PutRequest) returns (PutResponse) {}
    rpc Append (AppendRequest) returns (AppendResponse) {}

}

message GetRequest {
    string key = 1;
}
  
message GetResponse {
    string value = 1;
    string status = 2;
}

message PutRequest {
    string key = 1;
    string value = 2;
}
  
message PutResponse {
    string status = 1;
}

message AppendRequest {
    string key = 1;
    string args = 2;
}
  
message AppendResponse {
    string old_value = 1;
    string status = 2;
}

